src/components/PivotTable/PivotTableCell/PivotTableCell.test.tsx

```diff
@@ -0,0 +1,205 @@
+import React from 'react'
+import { fireEvent, getNodeText, render } from '@testing-library/react'
+import * as utilsModule from './utils'
+import { PivotTableCell, PivotTableCellProps } from './PivotTableCell'
+import { GridArea } from './classes/GridArea'
+import { PivotTableProvider, PivotTableContextType } from './PivotTableProvider'
+import { PivotTableCellTypes } from './model'
+
+const createComponent = ({ total, suffix }: PivotTableContextType, props: PivotTableCellProps) => {
+  return (
+    <PivotTableProvider value={{ total, suffix }}>
+      <PivotTableCell {...props} />
+    </PivotTableProvider>
+  )
+}
+
+describe('PivotTableCell', () => {
+  const [rowStart, columnStart] = [0, 0]
+  const gridArea = new GridArea(rowStart, columnStart)
+  const total = 1
+  const children = 1
+
+  it('should render correctly', () => {
+    const types = ['value'] as PivotTableCellTypes
+    const { container } = render(createComponent({ total }, { gridArea, types, children }))
+    expect(container).toMatchSnapshot()
+  })
+
+  it('should color font and background correctly', () => {
+    const types = ['value'] as PivotTableCellTypes
+    const expectedColor = 'rgb(0, 1, 2)'
+    const expectedBackgroundColor = 'rgb(2, 1, 0)'
+
+    const mockCalculateCellColor = jest.spyOn(utilsModule, 'calculateCellColor').mockReturnValue({
+      color: expectedColor,
+      backgroundColor: expectedBackgroundColor,
+    })
+
+    const { container } = render(createComponent({ total }, { gridArea, types, children }))
+    const { color, backgroundColor } = window.getComputedStyle(
+      selectPivotTableCellElement(container, rowStart, columnStart)
+    )
+
+    expect(color).toEqual(expectedColor)
+    expect(backgroundColor).toEqual(expectedBackgroundColor)
+    expect(mockCalculateCellColor).toBeCalled()
+  })
+
+  describe('hover', () => {
+    describe.each(['value', 'empty'])("when cell type includes '%s'", (typeValue) => {
```

Comment:
I think is better to add the `typeValue` type

Reply:
The type casting still is necessary 😢 Maybe change this type to an array of enum maybe fix this